### PR TITLE
Feature/manage course page

### PR DESCRIPTION
**Pull request by SayPls**

### CODE REVIEW

File: backend/Codi.Core/Codi.Core.BLL/Services/CourseService.cs [hunk 1]
```diff
@@ -98,4 +102,109 @@ public async Task LeaveCourseAsync(LeaveCourseDto leaveCourseDto)
...
+        return _mapper.Map<CourseDto>(course);
+    }
+
+    public async Task<CourseUserDto> InviteCourseUserAsync(InviteCourseUserDto inviteCourseUserDto)
+    {
+        var existedCourseUser = await _context.CourseUsers
+            .IgnoreQueryFilters()
+            .FirstOrDefaultAsync(u => u.UserId == inviteCourseUserDto.UserId && u.CourseId == inviteCourseUserDto.CourseId);
+
+        if (existedCourseUser is not null && existedCourseUser.IsDeleted == false)
```
Comment (tatianahutii) [inline backend/Codi.Core/Codi.Core.BLL/Services/CourseService.cs:128]:
```suggestion
        if (existedCourseUser is not null && !existedCourseUser.IsDeleted)
```

File: backend/Codi.Core/Codi.Core.BLL/Services/CourseService.cs [hunk 2]
```diff
@@ -98,4 +102,109 @@ public async Task LeaveCourseAsync(LeaveCourseDto leaveCourseDto)
...
+        if(existedCourseUser is not null && existedCourseUser.IsDeleted)
+        {
+            existedCourseUser.IsDeleted = false;
+            existedCourseUser.CourseRole = CourseRole.Member;
+            existedCourseUser.CreatedAt = DateTime.UtcNow;
+            var updatedCourseUser = _context.Update(existedCourseUser).Entity;
+            await _context.SaveChangesAsync();
+            return await GetInviteCourseUserByIdAsync(updatedCourseUser.UserId, updatedCourseUser.CourseId);
+
+        }
```
Comment (tatianahutii) [inline backend/Codi.Core/Codi.Core.BLL/Services/CourseService.cs:142]: You can just add filter by IsDeleted field when you load existedCourseUser

File: backend/Codi.Core/Codi.Core.Common/DTO/Course/CourseUserDto.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+﻿
+
```
Comment (tatianahutii) [inline backend/Codi.Core/Codi.Core.Common/DTO/Course/CourseUserDto.cs:2]: Please delete extra space

File: backend/Codi.Core/Codi.Core.Common/DTO/Course/CourseUserDto.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+using Codi.Core.Common.DTO.User;
+using Codi.Core.Common.Enums;
+
+
```
Comment (tatianahutii) [inline backend/Codi.Core/Codi.Core.Common/DTO/Course/CourseUserDto.cs:6]: Please delete extra space

File: frontend/src/app/core/models/course/course-role.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+export enum CourseRole {
+    Admin = 0,
+    Member = 1
```
Comment (tatianahutii) [inline frontend/src/app/core/models/course/course-role.ts:3]: You can don't write numbers here

File: frontend/src/app/modules/main/course-manage-page/course-invite-dialog/course-invite-dialog.component.html [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+                            <button ngbDropdownItem (click)="changeRole(courseUser.user.id, courseRole.Member)">{{courseRole[1]}}</button>
+                        </div>
+                    </div>
+                    <ng-template #forMember>
+                        {{courseRole[courseUser.courseRole]}}
+                    </ng-template>
+                </div>
+            </div>
+            <div>
+                <a type="button" style="color: #8834ee" *ngIf="!currentCourseUser.courseRole" (click)="deleteUserCourse(courseUser.user.id)" >&#10008;</a>
```
Comment (tatianahutii) [inline frontend/src/app/modules/main/course-manage-page/course-invite-dialog/course-invite-dialog.component.html:85]: Please do not use inline styles

File: frontend/src/app/modules/main/course-manage-page/course-invite-dialog/course-invite-dialog.component.ts [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+                    catchError(() => {
+                        this.searchFailed = true;
+                        return of([]);
+                    }))
+            ),
+            tap(() => this.searching = false)
+        );
+
+    searchUser(term: string) {
+        if (term === '') {
```
Comment (tatianahutii) [inline frontend/src/app/modules/main/course-manage-page/course-invite-dialog/course-invite-dialog.component.ts:76]:
```suggestion
        if (!term) {
```

File: frontend/src/app/modules/main/course-manage-page/course-invite-dialog/course-invite-dialog.component.ts [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+            .subscribe({
+                next:(courseUsers) => {
+                    this.courseUsers = courseUsers;
+                },
+                error:() => this.notificationService.showErrorMessage('Something went wrong', "Error")
+            });
+    }
+
+    inviteUser() {
+        let invitedUser: CourseUser;
```
Comment (tatianahutii) [inline frontend/src/app/modules/main/course-manage-page/course-invite-dialog/course-invite-dialog.component.ts:103]: Why do you need this variable?

File: frontend/src/app/modules/main/course-manage-page/course-invite-dialog/course-invite-dialog.component.ts [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+            courseId: this.course.id
+        } as InviteUserCourse;
+        this.form.reset();
+        this.courseService
+            .inviteUserToCourse(inviteUserToCourse)
+            .pipe(takeUntil(this.unsubscribe$))
+            .subscribe({
+                next:(resp) => {
+                    invitedUser = resp
+                    this.reloadCourseUsers( 1 )
```
Comment (tatianahutii) [inline frontend/src/app/modules/main/course-manage-page/course-invite-dialog/course-invite-dialog.component.ts:116]: Please get rid of this magic number

File: frontend/src/app/modules/main/course-manage-page/course-invite-dialog/course-invite-dialog.component.ts [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+            courseId: this.course.id,
+            courseRole: role.valueOf()
+        } as InviteUserCourse
+        this.courseService
+            .changeCourseUserRole(changeCourseUserRole)
+            .pipe(takeUntil(this.unsubscribe$))
+            .subscribe({
+                next:(resp) => {
+                    changedUserRole = resp
+                    this.reloadCourseUsers(1)
```
Comment (tatianahutii) [inline frontend/src/app/modules/main/course-manage-page/course-invite-dialog/course-invite-dialog.component.ts:137]: See the previous comment

File: frontend/src/app/modules/main/course-manage-page/course-invite-dialog/course-invite-dialog.component.ts [hunk 11]
```diff
@@ -0,0 +1,161 @@
...
+        let deleteUser = {
+            courseId: this.course.id,
+            userId: id
+        } as DeleteUserCourse
+        this.courseService
+            .deleteUserCourse(deleteUser)
+            .pipe(takeUntil(this.unsubscribe$))
+            .subscribe({
+                next:() => {
+                    this.reloadCourseUsers(1)
```
Comment (tatianahutii) [inline frontend/src/app/modules/main/course-manage-page/course-invite-dialog/course-invite-dialog.component.ts:154]: See the previous comment

File: frontend/src/app/modules/main/course-manage-page/course-manage-page.component.sass [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+
+.team-member-avatar
+    padding-right: 32px
+    display: flex
+
+.line
+    width: 100%
+    margin: 10px
+    &:after
+        content: ''
```
Comment (tatianahutii) [inline frontend/src/app/modules/main/course-manage-page/course-manage-page.component.sass:52]: Do you really need this?

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Comment by tatianahutii** [comment]
Could you please add screenshots? 😺

File: backend/Codi.Core/Codi.Core.BLL/Services/CourseService.cs (hunk 2 above)
Comment (SayPls) [reply backend/Codi.Core/Codi.Core.BLL/Services/CourseService.cs:142]: I ignore this filter for inviting previously deleted course users from current course

Comment (tatianahutii) [reply backend/Codi.Core/Codi.Core.BLL/Services/CourseService.cs:142]: But you need to filter users, and this field should not affect the result

**Comment by SayPls** [comment]
image
image
image

**Review by tatianahutii — APPROVED** [review]

**Comment by tatianahutii** [comment]
LGTM

**Pull request merged**